File: CareMate-Backend/app/models/schedule.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum


class TaskStatus(str, Enum):
    PENDING = "pending"
    IN_PROGRESS = "in-progress"
    COMPLETED = "completed"


class ScheduleTaskCreate(BaseModel):
    title_en: str
    title_my: str
    time: str  # "HH:MM"
    status: TaskStatus = TaskStatus.PENDING
    user_id: str


class ScheduleTaskUpdate(BaseModel):
    title_en: Optional[str] = None
    title_my: Optional[str] = None
    time: Optional[str] = None
    status: Optional[TaskStatus] = None


class ScheduleTaskOut(BaseModel):
    id: str
    title_en: str
    title_my: str
    time: str
    status: TaskStatus
    user_id: str
    created_at: datetime
    updated_at: datetime

File: CareMate-iOS/CareMate/Models/Schedule.swift
import Foundation

enum TaskStatus: String, Codable, CaseIterable {
    case pending    = "pending"
    case inProgress = "in-progress"
    case completed  = "completed"

    var next: TaskStatus {
        switch self {
        case .pending:    return .inProgress
        case .inProgress: return .completed
        case .completed:  return .pending
        }
    }

    var displayName: String {
        switch self {
        case .pending:    return "Pending"
        case .inProgress: return "In Progress"
        case .completed:  return "Completed"
        }
    }
}

struct ScheduleTask: Identifiable, Codable {
    var id: String
    var titleEn: String
    var titleMy: String
    var time: String
    var status: TaskStatus
    var userId: String
    var createdAt: Date?
    var updatedAt: Date?

    enum CodingKeys: String, CodingKey {
        case id
        case titleEn  = "title_en"
        case titleMy  = "title_my"
        case time, status
        case userId   = "user_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }

    func title(for language: String) -> String {
        language == "my" ? titleMy : titleEn
    }
}

struct ScheduleTaskCreate: Codable {
    var titleEn: String
    var titleMy: String
    var time: String
    var status: TaskStatus = .pending
    var userId: String

    enum CodingKeys: String, CodingKey {
        case titleEn = "title_en"
        case titleMy = "title_my"
        case time, status
        case userId  = "user_id"
    }
}
